File: src/main/java/astnodes/parameters/InParameter.java
package astnodes.parameters;

import astnodes.exp.var.MemLoad;
import astnodes.exp.var.Register;

public class InParameter extends Parameter {

    private MemLoad alias;

    public InParameter(Register name, Register register) {
        super(name, register);
    }

    public MemLoad getAlias() {
        return alias;
    }

    public void setAlias(MemLoad alias) {
        this.alias = alias;
    }
}

File: src/main/scala/astnodes/parameters/InParameter.scala
package astnodes.parameters

import astnodes.exp.variable.{MemLoad, Register}

case class InParameter(name: Register, register: Register) extends Parameter(name, register) {
  /*  
  var alias: MemLoad = null

  def getAlias: MemLoad = {
    alias
  }

  def setAlias(alias: MemLoad): Unit = {
    this.alias = alias
  }
  */
}